### PR TITLE
Refactor StringPrinter

### PR DESCRIPTION
**Pull request by Chemaclass**
### 🔖 Changes

- Add `readonly` to `Printer` and `StringPrinter` classes
- Use individual unit tests instead of dataProvider for `StringPrinterTest`

### CODE REVIEW

File: src/php/Printer/TypePrinter/StringPrinter.php [hunk 1]
```diff
@@ -128,18 +138,18 @@ private function isMask11110XXX(int $asciiChar): bool
...
-            return dechex((($a - 0xE0) << 12) + ((((int) $str[2]) - 0x80) << 6) + ((int) $str[3]) - 0x80);
+            return dechex((($a - 0xE0) << 12) + ((((int)$str[2]) - 0x80) << 6) + ((int)$str[3]) - 0x80);
         }
         if (0xC0 <= $a) {
-            return dechex((($a - 0xC0) << 6) + ((int) $str[2]) - 0x80);
+            return dechex((($a - 0xC0) << 6) + ((int)$str[2]) - 0x80);
         }
 
-        return (string) $a;
+        return (string)$a;
```
Comment (JesusValera) [inline src/php/Printer/TypePrinter/StringPrinter.php:152]: Could you rollback these changes?

https://www.php-fig.org/per/coding-style/#61-unary-operators

File: src/php/Printer/TypePrinter/StringPrinter.php [hunk 2]
```diff
@@ -30,6 +30,16 @@ public function __construct(
     ) {
     }
 
+    public static function nonReadable(bool $withColor = false): self
```
Comment (JesusValera) [inline src/php/Printer/TypePrinter/StringPrinter.php:33]: Having these new `nonReadable` and `readable` named-constructors, does it make sense to make `private` the constructor?

**Review by JesusValera — APPROVED** [review]
Small suggestions, but not very important 😄

**Comment by Chemaclass** [comment]
Thanks for the help, @JesusValera !!

**Pull request merged**